Fix build for Expo SDK 54

<!--
  Please describe your bug clearly and concisely.

  Please also leave a note on your environment, particularly the versions React Native and analytics-react-native used
-->

- `analytics-react-native` version:
- Integrations versions (if used):
- React Native version:
- iOS or Android or both?

**Steps to reproduce**
Using latest Expo SDK 54 which uses "react-native": "0.81.1" leads to build fail due to Segment lib:

````
> Task :segment_analytics-react-native:compileDebugKotlin FAILED
e: file:///Users/rohovdmytro/Projects/cohios/node_modules/@segment/analytics-react-native/android/src/main/java/com/segmentanalyticsreactnative/AnalyticsReactNativeModule.kt:192:9 Unresolved reference 'currentActivity'.
e: file:///Users/rohovdmytro/Projects/cohios/node_modules/@segment/analytics-react-native/android/src/main/java/com/segmentanalyticsreactnative/AnalyticsReactNativeModule.kt:197:32 Unresolved reference 'currentActivity'.
e: file:///Users/rohovdmytro/Projects/cohios/node_modules/@segment/analytics-react-native/android/src/main/java/com/segmentanalyticsreactnative/AnalyticsReactNativeModule.kt:220:19 Unresolved reference 'currentActivity'.
e: file:///Users/rohovdmytro/Projects/cohios/node_modules/@segment/analytics-react-native/android/src/main/java/com/segmentanalyticsreactnative/AnalyticsReactNativeModule.kt:252:9 Unresolved reference 'currentActivity'.
e: file:///Users/rohovdmytro/Projects/cohios/node_modules/@segment/analytics-react-native/android/src/main/java/com/segmentanalyticsreactnative/AnalyticsReactNativeModule.kt:254:37 Unresolved reference 'currentActivity'.
e: file:///Users/rohovdmytro/Projects/cohios/node_modules/@segment/analytics-react-native/android/src/main/java/com/segmentanalyticsreactnative/AnalyticsReactNativeModule.kt:254:55 Unresolved reference 'intent'.
e: file:///Users/rohovdmytro/Projects/cohios/node_modules/@segment/analytics-react-native/android/src/main/java/com/segmentanalyticsreactnative/AnalyticsReactNativeModule.kt:255:22 Unresolved reference 'currentActivity'.
e: file:///Users/rohovdmytro/Projects/cohios/node_modules/@segment/analytics-react-native/android/src/main/java/com/segmentanalyticsreactnative/AnalyticsReactNativeModule.kt:255:40 Unresolved reference 'intent'.

[Incubating] Problems report is available at: file:///Users/rohovdmytro/Projects/cohios/android/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':segment_analytics-react-native:compileDebugKotlin'.
> A failure occurred while executing org.jetbrains.kotlin.compilerRunner.GradleCompilerRunnerWithWorkers$GradleKotlinCompilerWorkAction
   > Compilation error. See log for more details

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/8.14.3/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 39s
````

**Expected behavior**
Build works.

**Actual behavior**
Build fails.

--- a/packages/core/android/src/main/java/com/segmentanalyticsreactnative/AnalyticsReactNativeModule.kt
+++ b/packages/core/android/src/main/java/com/segmentanalyticsreactnative/AnalyticsReactNativeModule.kt
@@ -189,12 +189,12 @@ class AnalyticsReactNativeModule : ReactContextBaseJavaModule, ActivityEventList
 
     val properties = Hashtable<String, String>()
 
-    if (currentActivity == null) {
+    if (reactApplicationContext.currentActivity == null) {
       Log.d(name, "No activity found")
       return
     }
 
-    val referrer = getReferrer(currentActivity!!)
+    val referrer = getReferrer(reactApplicationContext.currentActivity!!)
 
     if (referrer != null) {
       val referringApplication = referrer.toString()
@@ -217,14 +217,14 @@ class AnalyticsReactNativeModule : ReactContextBaseJavaModule, ActivityEventList
     }
 
     Log.d(name, "Sending Deeplink data to store: uri=${uri}, referrer=${referrer}")
-    val sovran = (currentActivity?.application as ReactApplication)
+    val sovran = (reactApplicationContext.currentActivity?.application as ReactApplication)
       ?.reactNativeHost
       ?.reactInstanceManager
       ?.currentReactContext
       ?.getNativeModule(SovranModule::class.java)
     sovran?.dispatch("add-deepLink-data", properties)
 
-    
+
   }
 
   fun setAnonymousId(anonymousId: String) {
@@ -249,10 +249,10 @@ class AnalyticsReactNativeModule : ReactContextBaseJavaModule, ActivityEventList
   }
 
   override fun onHostResume() {
-    if (currentActivity != null && isColdLaunch) {
+    if (reactApplicationContext.currentActivity != null && isColdLaunch) {
       isColdLaunch = false
-      Log.d(name, "onHostResume = ${currentActivity!!.intent}")
-      trackDeepLinks(currentActivity!!.intent)
+      Log.d(name, "onHostResume = ${reactApplicationContext.currentActivity!!.intent}")
+      trackDeepLinks(reactApplicationContext.currentActivity!!.intent)
     }
   }
 

